Spigot-Server-Patches/0599-Configurable-bamboo-natural-growth-heights.patch

```diff
@@ -0,0 +1,85 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Wed, 11 Nov 2020 17:47:37 -0800
+Subject: [PATCH] Configurable bamboo natural growth heights
+
+Both the minimum fully-grown heights and the maximum are configurable
+
+diff --git a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+index 4aacf8a5b2e98b0f8c1e27569ff7591fee32c725..281762d72f46bdfd30c154c12af1add6e377f208 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+@@ -74,10 +74,14 @@ public class PaperWorldConfig {
+ 
+     public int cactusMaxHeight;
+     public int reedMaxHeight;
++    public int bambooMaxHeight;
++    public int bambooMinHeight;
+     private void blockGrowthHeight() {
+         cactusMaxHeight = getInt("max-growth-height.cactus", 3);
+         reedMaxHeight = getInt("max-growth-height.reeds", 3);
+-        log("Max height for cactus growth " + cactusMaxHeight + ". Max height for reed growth " + reedMaxHeight);
++        bambooMaxHeight = getInt("max-growth-height.bamboo.max", 16);
++        bambooMinHeight = getInt("max-growth-height.bamboo.min", 10);
```

Comment:
Shouldn't this be 11?

Reply:
I took the min value directly from [Billy's patch on Purpur](https://github.com/pl3xgaming/Purpur/blob/ver/1.16.3/patches/server/0082-Implement-bamboo-growth-settings.patch).

I guess it does default to 11 initially.

Reply:
Yes, 11 is correct here. For some reason I changed `<` to `<=` and used 10 in my patch. I dont remember why now :3